<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form1</class>
 <widget class="QMainWindow" name="Form1">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(0, 0, 0);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QLabel" name="header">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <pointsize>25</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(255, 255, 255);</string>
      </property>
      <property name="text">
       <string>Расходимость лучей</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="text">
             <string>Прицельный параметр</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spin1">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="minimum">
             <double>-0.960000000000000</double>
            </property>
            <property name="maximum">
             <double>0.960000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.860000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="slider1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="minimum">
           <number>-96</number>
          </property>
          <property name="maximum">
           <number>96</number>
          </property>
          <property name="value">
           <number>86</number>
          </property>
          <property name="sliderPosition">
           <number>86</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label2">
            <property name="font">
             <font>
              <pointsize>12</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="text">
             <string>Длина волны</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spin2">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>25</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(255, 255, 255);</string>
            </property>
            <property name="minimum">
             <number>400</number>
            </property>
            <property name="maximum">
             <number>760</number>
            </property>
            <property name="value">
             <number>600</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QSlider" name="slider2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="minimum">
           <number>400</number>
          </property>
          <property name="maximum">
           <number>760</number>
          </property>
          <property name="value">
           <number>600</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Первичная радуга</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_2">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Вторичная радуга</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(170,170,170);</string>
          </property>
          <property name="text">
           <string>&amp;Сброс</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>68</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="quit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(170,170,170);</string>
          </property>
          <property name="text">
           <string>&amp;Выход</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="GLWidget" name="widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>quit</sender>
   <signal>clicked()</signal>
   <receiver>Form1</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>14</x>
     <y>582</y>
    </hint>
    <hint type="destinationlabel">
     <x>24</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton</sender>
   <signal>clicked()</signal>
   <receiver>Form1</receiver>
   <slot>radio1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>49</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Form1</receiver>
   <slot>radio2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Form1</receiver>
   <slot>wave_spin_d(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>172</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Form1</receiver>
   <slot>wave_slider_d(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>126</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin1</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Form1</receiver>
   <slot>dist_spin_d(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>441</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>slider1</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Form1</receiver>
   <slot>dist_slider_d(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>72</x>
     <y>419</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Form1</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>412</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>radio1()</slot>
  <slot>radio2()</slot>
  <slot>wave_spin_d(int)</slot>
  <slot>wave_slider_d(int)</slot>
  <slot>dist_spin_d(double)</slot>
  <slot>dist_slider_d(int)</slot>
  <slot>reset()</slot>
 </slots>
</ui>
